File: backend/HeartmadeCandles/HeartmadeCandles.Order.Core/Models/ConfiguredCandleFilter.cs
namespace HeartmadeCandles.Order.Core.Models;

public class ConfiguredCandleFilter
{
    public int CandleId { get; init; }

    public int? DecorId { get; init; }

    public int NumberOfLayerId { get; init; }

    public required int[] LayerColorIds { get; init; }

    public int? SmellId { get; init; }

    public int WickId { get; init; }

    public int Quantity { get; init; }

    public string? FilterString { get; init; }
}

File: frontend/heartmage-candles/src/typesV2/order/ConfiguredCandleFilter.ts
export interface ConfiguredCandleFilter {
  candleId: number;
  decorId: number;
  layerColorIds?: number[];
  numberOfLayerId: number;
  smellId?: number;
  wickId: number;
  quantity: number;
}

export const ParseToFilter = (orderItemFilter: ConfiguredCandleFilter): string => {
  let filter = `c-${orderItemFilter.candleId}`;
  filter += `~n-${orderItemFilter.numberOfLayerId}`;
  if (orderItemFilter.layerColorIds) {
    const layerColorsString = orderItemFilter.layerColorIds.join('_');
    filter += `~l-${layerColorsString}`;
  }
  if (orderItemFilter.decorId) {
    filter += `~d-${orderItemFilter.decorId}`;
  }
  if (orderItemFilter.smellId) {
    filter += `~s-${orderItemFilter.smellId}`;
  }
  if (orderItemFilter.wickId) {
    filter += `~w-${orderItemFilter.wickId}`;
  }
  filter += `~q-${orderItemFilter.quantity}`;

  return filter;
};
